src/dotnet/CarbonAware/Interfaces/ICarbonIntensityDataSource.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace CarbonAware.Interfaces;
+
+/// <summary>
+/// Represents a data source for carbon intensity.
+/// </summary>
+public interface ICarbonIntensityDataSource
+{
+    /// <summary>
+    /// Gets the carbon intensity for a given SciScoreCalculation.
+    /// </summary>
+    /// <param name="region">The region to get carbon data for.</param>
+    /// <param name="startPeriod">The start period.</param>
+    /// <param name="endPeriod">The end period.</param>
+    /// <returns>A list of emissions data for the given time period.</returns>
+    public Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(string region, DateTimeOffset startPeriod, DateTimeOffset endPeriod);
```

Comment:
I may be missing some context here but I'm confused about why we want to pass a region as a string? Seems like it might be better suited to be a location object. If lat/long is passed, or Ba is passed, or a string location, wouldn't those all be supported?

Reply:
I've updated this to take a location object.  BA is a watttime specific concept, so we're not allowing it to be supplied.  Instead, we're allowing Lattitude/Longitude, Azure region, Aws Region.  Does that work for what you're expecting?

Reply:
@rakker91 , understand that we can have multiple interfaces with different goals. My comment was more towards integration with upper layers (i.e aggregator and consumer (i.e webapi)) that can just flip a switch (env, config param) and start using a different plugin w/o re-implementing the aggregator & consumer stack.